Improve undergassing warnings (#377)
aptos-labs/developer-docs: apps/nextra/pages/en/build/advanced-guides/randomness.mdx

- `let winner_idx = aptos_framework::randomness::u64_range(0, n);` is the a randomness API call that returns a u64 integer in range `[0, n)` uniformly at random.
- `#[randomness]` is a required attribute to enable the API call at runtime.

> Note: there are security nuances to consider, explained in more details in "Security Considerations" section. Compiler helps with test and abort attacks by requiring functions using randomness to be private. On the other hand, randomness API currently doesn’t prevent undergassing attacks, and so code needs to be written in a specific way to avoid it.

Reviewer: ```suggestion
<Callout type="info">
### Security Considerations
Compiler helps with test and abort attacks, requiring functions using randomness to be private.  However, the randomness API current does not prevent undergassing attacks.  The smart contract will be need to be written in a certain way to avoid it.
</Callout>
```

Here are some ideas of how to prevent undergasing attack generally.

- Make your entry function gas independent from the randomness outcome.

>   Simplest example is to not “act” on the randomness outcome, i.e. read it and store it for later. Note that calling any framework functions or functions not completely understood can have variable gas costs. For example, calling randomness to decide which of the two players should get the winnings, and then depositing winnings to the target address might seem like a fixed gas cost, but coin::transfer / fungible_asset::transfer can have variable cost based on the user's onchain state.

Reviewer: ```suggestion
  The simplest example is to not "act" on the randomness outcome, i.e. read it and store it for later. Note that calling any other functions  can have variable gas costs. For example, when calling randomness to decide which player should win, and then depositing the winnings to the winner might seem like a fixed gas cost. But, `0x1::coin::transfer` / `0x1::fungible_asset::transfer` can have a variable cost based on the user's onchain state.
```
